tests/deployment/sagemaker/test_sagemaker.py

```diff
@@ -1,33 +1,419 @@
+from __future__ import absolute_import
+
+import botocore
+import pytest
+from mock import patch, MagicMock, mock_open
+from sys import version_info
+
 from botocore.exceptions import ClientError
 from botocore.stub import Stubber
 
 import boto3
+from moto import mock_ecr, mock_iam, mock_sts
 
-from bentoml.deployment.sagemaker import _parse_aws_client_exception_or_raise
+from bentoml.deployment.sagemaker import (
+    _parse_aws_client_exception_or_raise,
+    _cleanup_sagemaker_model,
+    _cleanup_sagemaker_endpoint_config,
+    get_arn_role_from_current_aws_user,
+    SageMakerDeploymentOperator,
+)
+from bentoml.proto.deployment_pb2 import Deployment
+from bentoml.proto.repository_pb2 import Bento, BentoServiceMetadata, GetBentoResponse
 from bentoml.proto.status_pb2 import Status
+from tests.deployment.sagemaker.sagemaker_moto import moto_mock_sagemaker
+
 
+@pytest.fixture(scope='function')
+def sagemaker_client():
+    return boto3.client('sagemaker', 'us-west-2')
 
-def test_sagemaker_handle_client_errors():
-    client = boto3.client('sagemaker', 'us-west-2')
-    stubber = Stubber(client)
+
+def test_sagemaker_handle_client_errors(sagemaker_client):
```

Comment:
this is enssentially a unit test for function ```_parse_aws_client_exception_or_raise```, make it simpler without the mock staff:
```
err = ClientError(
           error_response={
               'Error': {'Code': 'MyCode', 'Message': 'MyMessage'}},
           operation_name='myoperation'
         )
result = _parse_aws_client_exception_or_raise(err)
assert result....

```